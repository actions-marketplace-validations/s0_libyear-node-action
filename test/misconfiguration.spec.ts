import { runAction, expectRunError } from './util';

describe('misconfiguration', () => {
  it('missing GitHub env vars', async () => {
    await expect(
      runAction({
        eventJson: 'missing-var',
        logToConsole: false,
      })
    ).rejects.toThrowError('Invalid config, unable to continue');
  });

  it('missing event file', async () => {
    await expect(
      runAction({
        eventJson: 'missing-file',
        eventName: 'push',
        logToConsole: false,
      })
    ).rejects.toThrowError('ENOENT: no such file or directory');
  });

  it('unsupported event', async () => {
    await expectRunError(
      runAction({
        eventJson: {},
        eventName: 'foo',
        logToConsole: false,
      }),
      (err) => {
        expect(err.output.stderr).toMatchSnapshot();
      }
    );
  });
});
