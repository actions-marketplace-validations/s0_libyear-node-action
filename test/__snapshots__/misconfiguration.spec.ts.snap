// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`misconfiguration unsupported event 1`] = `
"::error ::Unsupported GitHub event: foo
"
`;
